modules/misunderstood/src/views/full/MainScreen/ChatPreview.tsx

```diff
@@ -1,11 +1,22 @@
+import ReactMessageRenderer, { defaultMessageConfig } from '@botpress/messaging-components'
 import classnames from 'classnames'
 import React from 'react'
 
 import { ContextMessage } from '../../../types'
-import { renderUnsafeHTML } from '../../lite/utils'
 
 import style from './style.scss'
 
+// TODO: Fix this either when we create the event or in the components lib
+const fixPayload = (payload: any): ContextMessage['payload'] => {
+  // Channel-web sends single-choice events and messaging replies with quick_reply
+  // We convert them into text elements since we only want to display the text value
+  if (payload.type === 'single-choice' || payload.type === 'quick_reply') {
+    payload = { ...payload, type: 'text' }
+  }
+
+  return payload
+}
+
```

Comment:
@charlescatta, @samuelmasse. Should we convert the choice component to be single-choice instead of quick_reply so that we only have to "fix" one type here?